### PR TITLE
Add `--clean` option for `bencher`

### PR DESCRIPTION
**Pull request by criccomini**
`bencher` now has a `--clean` argument that will delete all objects in the input path after the `bencher` subcommand runs. This switch works for both the compaction benchmarks and the db benchmarks.

By default, `--clean` is disabled since it's destructive. If `--clean` is set, `bencher` will check that the object store path is empty. If it is, it'll create a `.clean_benchmark_data` file, which will signal that the data should be deleted after the benchmark runs. If the path is not empty, the benchmark will fail. After the benchmark runs, if `--clean` is set and a `.clean_benchmark_data` exists, `bencher` will remove all objects from the path.

### CODE REVIEW

File: src/admin.rs [hunk 1]
```diff
@@ -56,6 +57,24 @@ pub async fn list_checkpoints(
...
+pub async fn delete_objects_with_prefix(
+    object_store: Arc<dyn ObjectStore>,
+    prefix: &Path,
+) -> Result<(), Box<dyn Error>> {
+    let stream = object_store
+        .list(Some(prefix))
+        .map_ok(|m| m.location)
+        .boxed();
+    object_store
+        .delete_stream(stream)
```
Comment (agavra) [inline src/admin.rs:71]: this is a little scary if you accidentally pass in the wrong path 😬 I know this is how most of these tools work, but do we want to have some kind of dummy file we create that indicates the dir was created by the bencher and then check for the presence of that file before running the delete?

File: src/admin.rs [hunk 2]
```diff
@@ -196,3 +215,69 @@ pub fn load_azure() -> Result<Arc<dyn ObjectStore>, Box<dyn Error>> {
...
     Ok(Arc::new(builder.build()?) as Arc<dyn ObjectStore>)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use object_store::{memory::InMemory, path::Path};
+
+    #[tokio::test]
+    async fn test_delete_objects_with_prefix_empty() {
```
Comment (agavra) [inline src/admin.rs:225]: can we add a test where an empty prefix deletes things?

File: src/bencher/benchmark-db.sh [hunk 3]
```diff
@@ -25,13 +25,13 @@ run_bench() {
   local log_file="$3"
 
   local bench_cmd="cargo run -r --bin bencher --features=bencher -- \
-    --path /slatedb-bencher_${put_percentage}_${concurrency} db \
+    --path /slatedb-bencher_${put_percentage}_${concurrency} --clean db \
```
Comment (agavra) [inline src/bencher/benchmark-db.sh:28]: any chance we can parameterize the `benchmark-db.sh` script on this? I've actually found this super helpful to examine the results of the bencher after it runs as a learning exercise

**Review by agavra — APPROVED** [review]
LGTM, a few smaller questions and one around maybe making this operation a litttttle safer

File: src/admin.rs (hunk 1 above)
Comment (criccomini) [reply src/admin.rs:71]: Good idea. WDYT about creating and checking the dummy file in main.rs and leaving this file as-is so it's more generic?

Comment (agavra) [reply src/admin.rs:71]: 👍 yeah definitely wouldn't make sense in the admin, I like your suggestion
